src/test/java/teammates/test/cases/ui/browsertests/GodModeTest.java

```diff
@@ -18,132 +18,116 @@
+        output.write(content);
+        output.close();
     }
 
     @Test
-    public void testGodMode() throws Exception {
-        if (!TestProperties.inst().isDevServer()) return;
-        
-        ______TS("test verifyHtml");
+    public void testGodMode() throws IOException {
         
         System.clearProperty("godmode");
-        Assumption.assertNull(System.getProperty("godmode"));
+        assertNull(System.getProperty("godmode"));
         
-        try {
-            // should fail as the expected file "godmodeOutpout
-            page.verifyHtml("/godmodeOutput.html");
-            signalFailureToDetectException(" - Assertion Error");
-        } catch (AssertionError ae) {
-            // expected
-        }
+        ______TS("test verifyHtml");
         
-        System.setProperty("godmode", "true");
-        // automatically generates the file and hence passes
-        page.verifyHtml("/godmodeOutput.html");
+        testGodMode(true);
         
-        System.clearProperty("godmode");
-        Assumption.assertNull(System.getProperty("godmode"));
+        ______TS("test verifyHtmlMainContent");
         
-        // should pass without need for godmode
-        // as the file has already been generated
-        page.verifyHtml("/godmodeOutput.html");
+        testGodMode(false);
         
-        String expectedOutputPage = FileHelper.readFile(getExpectedOutputFilePath());
-        String actualOutputPage = FileHelper.readFile(getOutputFilePath());
+    }
+    
+    private void testGodMode(boolean isPart) throws IOException {
+        // run the God mode with non-existent expected file
+        runGodModeRoutine(isPart);
         
-        // ensure that the generated file is as expected
-        verifyOutput(expectedOutputPage, actualOutputPage);
+        writeToFile(OUTPUT_PATH, PLACEHOLDER_CONTENT);
         
-        ______TS("test verifyHtmlMainContent");
-        // repeat the same procedure as above for this
+        // run the God mode with wrong content in expected file
+        runGodModeRoutine(isPart);
         
-        writeToFile(getOutputFilePath(), 
-                "<div id='mainContent'>test</div>");
+        // delete the output file generated
+        File file = new File(OUTPUT_PATH);
+        if (!file.delete()) {
+            print("Delete failed: " + file.getAbsolutePath());
+            file.deleteOnExit();
+        }
+    }
+    
+    private void runGodModeRoutine(boolean isPart) throws IOException {
         
         try {
-            page.verifyHtml("/godmodeOutput.html");
-            signalFailureToDetectException(" - Assertion Error");
-        } catch (AssertionError ae) {
-            // expected
+            // should fail as the expected file "godmodeOutput.html"
+            // either does not exist or has the wrong content
+            verifyHtml(OUTPUT_FILENAME, isPart);
+            signalFailureToDetectException();
+        } catch (IOException|AssertionError e) {
```

Comment:
I think to make this stronger, we should know the which exception is expected and only ignore that one